articles/active-directory/develop/quickstarts/active-directory-javascriptspa.md

```diff
@@ -0,0 +1,159 @@
+ms.workload: identity�
+ms.date: 05/09/2018
+ms.author: andret
+
+---
+
+# Sign-in users and acquire an access token from a JavaScript application
+
+This quickstart contains a code sample that demonstrates how a JavaScript Single Page Application (SPA) can sign in personal, work and school accounts, get an access token, and call the Microsoft Graph API.
+
+![How the sample app generated by this Quickstart works](media/active-directory-javascriptspa/javascriptspa-intro.png)
+
+> [!div renderon="docs"]
+> ## Register your application and download your Quickstart app
+> You have two options to start your Quickstart application:
+> ### Option 1: Register your application and download your Quickstart app [Express]
+>
+> 1. Go to the [Azure Portal - Application Registration [Prod]](https://portal.azure.com/signin/index/?Microsoft_AAD_RegisteredApps=true#blade/Microsoft_AAD_RegisteredApps/applicationsListBlade/quickStartType/JavascriptSpaQuickstartPage/sourceType/docs)
+> 1. Enter a name for your application and click **Register**
+> 1. Follow the instructions to download and configure your new application, adding **http<span/>://localhost:30662/** as Redirect URL and setting the **Implict Flow** automatically for you.
+> [!div renderon="docs"]
+> ### Option 2: Register your application and configure sample [Manual]
+> #### Step 1: Register your application
+> 
+> 1. To register an application, go to the [Azure Portal - Application Registration [Prod]](https://aka.ms/registeredappsprod) and select **New registration**
+> 1. Enter a name for your application, add **http<span/>://localhost:30662/** in Reply URL, and click **Register**
+> 1. Select **Authentication** menu and set **ID tokens** under *Implict Grant* and select 'Save'
+
+> [!div class="sxs-lookup" renderon="portal"]
+> ### Step 1: Configure your application in Azure Portal
+> For the code sample for this quickstart to work, you need to add a reply URL as **http<span/>://localhost:30662/** and enable **Implict Grant**.
+> > [!div renderon="portal" id="makechanges" class="nextstepaction"]
+> > [Make these changes for me]()
+>
+> > [!div id="appconfigured" class="hidden"]
+> > ![Already configured](media/active-directory-windesktop/checkmark.png) Your application is configured with these attributes
+
+> [!NOTE]
+> If you use Visual Studio, the Redirect URL will be set to to *http:<span/>//localhost:30662/* as it is configured in the code sample's project. If you use Python or any other web server, set redirect URL to *http://<span/>localhost:30662/*, run the following command line:
+>> ```bash
+>> python -m http.server 30662
+>> ```
+
+### Step 2: Download your web server or project
+
+- [Download the Visual Studio project](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/VisualStudio.zip)
+- [Download the core project files - for a local web server, such as Python](https://github.com/Azure-Samples/active-directory-javascript-graphapi-v2/archive/core.zip)
```

Comment:
why do we need 2 samples?

Reply:
@asnow003 , @nehaagrawal - the reason is that our audience is disproportionally skewed towards Visual Studio audiences. Having a Visual Studio project ready-to-go helps people that use Visual Studio. In fact, on our [Guided Setup](https://docs.microsoft.com/azure/active-directory/develop/guidedsetups/active-directory-javascriptspa), clickthrough  rates on the Visual Studio option is more than 100% over the *non-Visual Studio* version. I believe having a Visual Studio option in addition to just the project files provides a better user experience

Reply:
This is a typescript library and users will need Visual studio only if their web service is written in Visual studio. If we are demonstrating the use of MS Graph in our sample, it doesn't require Visual Studio. I am still not convince why we need visual studio.

Reply:
VS is just an option to make sure VS users hava an easy path to build and run: for example, if I am a VS user, I can download a zip file, extract it, load in Visual Studio, configure, press <F5> and experiement my app. IF we don't have a VS option, a VS developer would need to do several steps to run the sample app - having an option make it easier for VS users - and like the number suggested, this is the preffered option for most of the users

Reply:
After reading all the feedback and response from Andre, I think we should offer two options, in this order
1. using node.js
2. using visual studio

Reply:
I agree with Manoj's feedback here. Even if we offer Visual Studio option, there should be a way for non VS developers to build and run the sample code without having to deal with VS dependency. Node.js does provide a local web server and seems suitable for JS community than Python.